ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/image/ImageComponentView.kt

```diff
@@ -0,0 +1,138 @@
+package com.revenuecat.purchases.ui.revenuecatui.components.image
+
+import android.graphics.Color
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.paywalls.components.properties.ColorInfo
+import com.revenuecat.purchases.paywalls.components.properties.Size
+import com.revenuecat.purchases.paywalls.components.properties.SizeConstraint
+import com.revenuecat.purchases.ui.revenuecatui.components.modifier.size
+import com.revenuecat.purchases.ui.revenuecatui.components.style.ImageComponentStyle
+import com.revenuecat.purchases.ui.revenuecatui.composables.RemoteImage
+import com.revenuecat.purchases.ui.revenuecatui.extensions.applyIfNotNull
+import java.net.URL
+import androidx.compose.ui.graphics.Color as ComposeColor
+
+@Suppress("SpreadOperator", "LongParameterList")
+@JvmSynthetic
+@Composable
+internal fun ImageComponentView(
+    style: ImageComponentStyle,
+    modifier: Modifier = Modifier,
+) {
+    if (style.visible) {
+        Box(modifier = modifier.applyIfNotNull(style.shape) { clip(it) }) {
+            RemoteImage(
+                urlString = style.urlToUse().toString(),
+                modifier = Modifier
+                    .size(style.size)
+                    .applyIfNotNull(style.gradientColors()) { gradientColors ->
+                        drawWithCache {
+                            val gradientBrush: Brush? = when (gradientColors) {
+                                is ColorInfo.Gradient.Radial -> {
+                                    Brush.radialGradient(*gradientColors.points.toColorStops())
+                                }
+                                is ColorInfo.Gradient.Linear -> {
+                                    // TODO-PAYWALLS-V2: Support degrees
```

Comment:
It's a bit tricky to support degrees in a linear gradient... Since in iOS it seems we only support vertical gradients I thought to leave it like this for now and come back to it later.

Reply:
If you specify a `ColorStyle` instead of `ColorInfo` in `ImageComponentStyle`, you get the correct brush right away, including degrees support (*). There's already a `ColorScheme.toColorStyle()` transformation method. 

I think that also means you can consolidate `gradientColors` and `darkGradientColors`. (Maybe it can just be named `colorOverlay`, as I believe we support solid colors as well?)

(*): degrees support is the same as on iOS right now. We want to tweak it slightly on both platforms. See [this Slack discussion](https://revenuecat.slack.com/archives/C07H5S8CFL3/p1732716458417749?thread_ts=1732554856.424879&cid=C07H5S8CFL3).

Reply:
Ohhh nice! I totally missed that, thanks for the heads up!